data-access/src/object_store/mod.rs

```diff
@@ -80,15 +83,42 @@ pub trait ObjectStore: Sync + Send + Debug {
         prefix: &str,
         suffix: &str,
     ) -> Result<FileMetaStream> {
-        let file_stream = self.list_file(prefix).await?;
-        let suffix = suffix.to_owned();
-        Ok(Box::pin(file_stream.filter(move |fr| {
-            let has_suffix = match fr {
-                Ok(f) => f.path().ends_with(&suffix),
-                Err(_) => true,
```

Comment:
@tustvold -> original code where the errors were allowed to pass (Would be surprised to see anyone doing something meaningful with those..)

Reply:
try_filter doesn't filter out errors, see https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f7f679b078eeaf042970fc4ce212640e

So this could be simplified to something like

```
use futures::future::ready;
use futures::stream::TryStreamExt;

...

file_stream.try_filter(|f| ready(f.path().end_with(&suffix)))
```

Reply:
Correct. 

Should have read the docs a bit better, clearly stated there: https://docs.rs/futures/0.3.1/futures/stream/trait.TryStreamExt.html#method.try_filter -> All errors are passed through without filtering in this combinator. Learning something new every day :)

Reply:
Applied the change as you suggested. Thanks for your patience with me :)